backend/src/common/util/service.go

```diff
@@ -95,18 +101,59 @@ func GetKubernetesClientFromClientConfig(clientConfig clientcmd.ClientConfig) (
 	return clientSet, config, namespace, nil
 }
 
-func GetRpcConnectionWithTimeout(address string, timeout time.Time) (*grpc.ClientConn, error) {
+func GetRpcConnectionWithTimeout(address string, tlsEnabled bool, caCertPath string, timeout time.Time) (*grpc.ClientConn, error) {
+	creds := insecure.NewCredentials()
+	if tlsEnabled {
+		if caCertPath == "" {
```

Comment:
I don't think this should be an error. If you don't specify the CA, it should fallback to use the system trusted CAs. For example, if the certs are generated by Let's Encrypt, you don't need additional CAs. The system trusted CAs are sufficient.

Please make this change in other places as well.